tests/ui/components/navigation/navigationcomponent.js

```diff
@@ -102,3 +112,70 @@ describe('navigation component configuration', () => {
     expect(tab).toMatchObject(expected);
   });
 });
+
+describe('navigation tab links are correct', () => {
+  let COMPONENT_MANAGER;
+
+  const defaultConfig = {
+    container: '#test-component',
+    verticalKey: 'verticalKey',
+    verticalPages: [
+      { verticalKey: 'vertical', url: './vertical', label: 'Vertical' },
+      { verticalKey: 'facets', url: './facets', label: 'Facets' }
+    ]
+  };
+
+  beforeEach(() => {
+    COMPONENT_MANAGER = mockManager({
+      globalStorage: new GlobalStorage(),
+      persistentStorage: new PersistentStorage()
+    });
+
+    COMPONENT_MANAGER.getComponentNamesForComponentTypes = () => [];
+  });
+
+  it('tab links contain the referrerPageUrl from global storage', () => {
+    COMPONENT_MANAGER.core.globalStorage.set(StorageKeys.REFERRER_PAGE_URL, 'yext.com');
+
+    const component = COMPONENT_MANAGER.create('Navigation', defaultConfig);
+    const wrapper = mount(component);
+
+    const navItem = wrapper.find('.js-yxt-navItem').first();
+    const tabLink = navItem.prop('href');
+
+    expect(tabLink).toContain('referrerPageUrl=yext.com');
+  });
+
+  it('tab links contain the context from global storage', () => {
+    COMPONENT_MANAGER.core.globalStorage.set(StorageKeys.API_CONTEXT, 'some context');
+
+    const component = COMPONENT_MANAGER.create('Navigation', defaultConfig);
+    const wrapper = mount(component);
+
+    const navItem = wrapper.find('.js-yxt-navItem').first();
+    const tabLink = navItem.prop('href');
+
+    expect(tabLink).toContain('context=some+context');
+  });
+
+  it('tab links default to the tab order from global storage', () => {
```

Comment:
The back-end can return a new tab order as part of a vertical search response. Could we have a test that verifies the component re-renders properly when a new tab order comes in after it's mounted?

Reply:
Yes, I was thinking of making that an acceptance test. Also, do you have an example query that re-orders the tabs on the slapshot test account? I wasn't able to find one

Reply:
I don't have one at hand. @creotutar might be more familiar. I believe the response just sends verticals in a particular order. We then set that order as the `tabOrder`. I think the relevant code is here: https://github.com/yext/answers/blob/master/src/core/models/navigation.js. The storage key is `StorageKeys.NAVIGATION`.

Reply:
For now, it may be sufficient to just write a Jest test that verifies that behavior. Mount the component, update `StorageKeys.NAVIGATION`, and verify the links re-render properly.

Reply:
I added a tab order test